Documentation/learning/glossary.md

```diff
@@ -2,22 +2,70 @@
 
 This document defines the various terms used in etcd documentation, command line and source code.
 
-## Node
+## Alarm
 
-Node is an instance of raft state machine.
+Alarms are a means to recognize any fault in the etcd server.
 
-It has a unique identification, and records other nodes' progress internally when it is the leader.
+## Authentication
 
-## Member
+Authentication is used to authenticate a user before the user can perform a set of operations.
 
-Member is an instance of etcd. It hosts a node, and provides service to clients.
+## Client
+
+Client is a caller of the cluster's HTTP API.
 
 ## Cluster
 
 Cluster consists of several members.
 
 The node in each member follows raft consensus protocol to replicate logs. Cluster receives proposals from members, commits them and apply to local store.
 
+## Command
+
+Command is an identifiers which are used to perform a specific operation in etcd server using etcdctl.
+Example: put, get, del, lease, etc.
+
+## Compaction
+
+Compaction discards all etcd event history prior to a given revision. Since etcd uses a multiversion concurrency control model, it preserves all key updates as event history. When the event history up to some revision is no longer needed, all superseded keys may be compacted away to reclaim storage space in the etcd backend database.
+
+## Election
+
+Election is an in-build functionality of etcd-server which facilitates in electing a leader in etcd server cluster.
+
+## Endpoint
+
+Endpoint refers to the target member where the client request or operation needs to be done.
+
+## Keys
+
+Keys are unique idetifiers. Keys are used to store values to the store and the values can be retrieved from the store using the keys.
+
```

Comment:
probably should define ranges too since they're used everywhere

```
## Key range

A set of keys containing either an individual key, a lexical interval for all x such that a < x <= b, or all keys greater than a given key.
```